crates/swift-integration-tests/src/async_function.rs

```diff
@@ -38,6 +38,30 @@ mod ffi {
         fn new(val: u32) -> AsyncResultOpaqueRustType2;
         fn val(&self) -> u32;
     }
+
+    enum AsyncResultOkEnum {
```

Comment:
Do we need new async specific types here? 

I'm thinking that we can just re-use the existing opaque Rust types and enum Result types and we don't need `AsyncResult*` opaque types or enums.

Reply:
There were no transparent enums in async_function.rs. So, I appended new ones.

Reply:
Ohhhh ok